sdk/core/core-lro/test/lro.spec.ts

```diff
@@ -942,538 +1338,418 @@ matrix([["createPoller", "LroEngine"]] as const, async function (implName: Imple
+        });
 
-          it("should handle postAsyncResourceLocation", async () => {
-            const locationPath = "/postlocation/retry/succeeded/operationResults/foo/200/";
-            const pollingPath = "/postlocation/retry/succeeded/operationResults/200/";
-            const result = await runLro({
+        it("should handle post202NonRetry400 ", async () => {
+          const path = `/nonretryerror/post/202/retry/400`;
+          await assertDivergentBehavior({
+            op: runLro({
               routes: [
                 {
                   method: "POST",
+                  path,
                   status: 202,
+                  body: `{ "properties": { "provisioningState": "Creating"}, "id": "100", "name": "foo" }`,
                   headers: {
-                    [headerName]: pollingPath,
-                    "retry-after": "0",
+                    Location: path,
+                    "Retry-After": "0",
                   },
-                  body: `{"status":"Accepted"}`,
                 },
                 {
                   method: "GET",
-                  path: pollingPath,
-                  status: 202,
-                  headers: {
-                    location: locationPath,
-                    [headerName]: pollingPath,
-                    "retry-after": "0",
-                  },
-                  body: `{"status":"Accepted"}`,
+                  path,
+                  status: 400,
+                  body: `{ "message" : "Expected bad request message" }`,
                 },
+              ],
+            }),
+            throwOnNon2xxResponse,
+            throwing: {
+              statusCode: 400,
+            },
+            notThrowing: {
+              messagePattern: /The long-running operation has failed/,
+            },
+          });
+        });
+
+        it("should handle postAsyncRelativeRetry400 ", async () => {
+          const pollingPath = `/nonretryerror/postasync/retry/failed/operationResults/400`;
+          await assertDivergentBehavior({
+            op: runLro({
+              routes: [
                 {
-                  method: "GET",
-                  path: pollingPath,
-                  status: 200,
-                  body: `{"status":"Succeeded", "resourceLocation": "${locationPath}"}`,
+                  method: "POST",
+                  status: 202,
+                  body: `{ "properties": { "provisioningState": "Creating"}, "id": "100", "name": "foo" }`,
+                  headers: createDoubleHeaders({
+                    pollingPath,
+                    headers: { "Retry-After": "0" },
+                  }),
                 },
                 {
                   method: "GET",
-                  path: locationPath,
-                  status: 200,
-                  body: `{"id":"100","name":"foo"}`,
+                  path: pollingPath,
+                  status: 400,
+                  body: `{ "message" : "Expected bad request message" }`,
```

Comment:
```suggestion
                  body: `{ "message" : "Expected bad request message" , "status" : 400}`,
```

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-js/pull/23296/commits/d13631a917b8dde83bc75f23797d55a1c2f57f3a